server/src/main/java/org/opensearch/search/aggregations/bucket/terms/InternalTerms.java

```diff
@@ -297,9 +297,11 @@ private BucketOrder getReduceOrder(List<InternalAggregation> aggregations) {
         return thisReduceOrder != null ? thisReduceOrder : order;
     }
 
-    private long getDocCountError(InternalTerms<?, ?> terms) {
+    private long getDocCountError(InternalTerms<?, ?> terms, ReduceContext reduceContext) {
         int size = terms.getBuckets().size();
-        if (size == 0 || size < terms.getShardSize() || isKeyOrder(terms.order)) {
+        // shard_size is not applied at the slice level, meaning no buckets are eliminated,
+        // therefore doc_count_error must also be 0 at the slice level.
+        if (reduceContext.isSliceLevel() || size == 0 || size < terms.getShardSize() || isKeyOrder(terms.order)) {
```

Comment:
`docCountError` is always computed at coordinator based on the buckets returned by the shards. I think to account for that we will need to return 0 even for partial reduce case which happens on shard (as part of concurrent search). Not very clear on the comment here.

Reply:
What I am trying to convey with the code comment is the doc_count_error comes from applying either `size` or `shard_size` parameters and eliminating candidate buckets, so because we do not eliminate any candidate buckets at the shard level partial reduce, we should always return 0 when `reduceContext.isSliceLevel() == true`.